### ISSUE TITLE
Better ec2 installation of correct parsl version

### ISSUE THREAD

**Issue by benclifford** [Parsl/parsl#1137]
At present, the ec2/aws template.py installs the latest release version of parsl using pip.

When running from development source, this often installs an incompatible version of parsl.

This is a problem for any users who attempt our very common "you have a bug; please use the latest `master` version of parsl" approach; and makes development and testing much harder.

A better way to install the same version of source as is being submitted from would be nice. For example, @annawoodard has commented on slack about how some condor-based installations install a worker-side parsl that is a copy of the submit-side python install tree.

**Comment by benclifford** [comment]
After some experimentation, I would be more comfortable if it was left to either the base AMI or the worker init to install the correct parsl version, rather than the template having a hard-coded often-incorrect install. The choice of base AMI and worker init is more easily configured by a user.

**Comment by yadudoc** [comment]
I agree, both AMI and worker_init are far more easier for the user. We should remove the whole bunch of installs around here: https://github.com/Parsl/parsl/blob/94de10275a0190fd7e59c5d5c2261fb81feeab72/parsl/providers/aws/template.py#L6

Assuming debian/ubuntu is another poor choice that we've had at least 1 user complain about.
